inc/zoo/swar/associative_iteration.h

```diff
@@ -218,16 +218,69 @@ constexpr auto multiplication_OverflowUnsafe_SpecificBitCount(
 
     auto halver = [](auto counts) {
         auto msbCleared = counts & ~S{S::MostSignificantBit};
-        return S{msbCleared.value() << 1};
+        return S{static_cast<T>(msbCleared.value() << 1)};
     };
 
-    multiplier = S{multiplier.value() << (NB - ActualBits)};
+    multiplier = S{static_cast<T>(multiplier.value() << (NB - ActualBits))};
     return associativeOperatorIterated_regressive(
         multiplicand, S{0}, multiplier, S{S::MostSignificantBit}, operation,
         ActualBits, halver
     );
 }
 
+
+/*
+     // extended from mathematics to generic programming
+     // see https://github.com/jamierpond/fmtgp/blob/main/2_first_algo/main.cpp
+
+      template <typename T> constexpr T exp_acc(T r, T a, T n) {
+        for (;;) {
+          if (is_odd(n)) {
+            r = multiply(r, a);
+            if (n == 1) {
+              return r;
+            }
+          }
+          n = half(n);
+          a = multiply(a, a);
+        }
+      }
+*/
+
+template<int ActualBits, int NB, typename T>
+constexpr auto expo_OverflowUnsafe_SpecificBitCount(
+    SWAR<NB, T> x,
+    SWAR<NB, T> exponent
+) {
+    using S = SWAR<NB, T>;
+
+    auto operation = [](auto left, auto right, auto counts) {
+      const auto mask = makeLaneMaskFromMSB(counts);
+      const auto antiMask = ~mask;
+      const auto product =
+        multiplication_OverflowUnsafe_SpecificBitCount<ActualBits>(left, right);
+      /*
+       * if (count)
+       *    return product;
+       * else
+       *    return left;
+       */
+      return (product & mask) | (left & antiMask);
+    };
+
+    // halver should work same as multiplication... i think...
+    auto halver = [](auto counts) {
+        auto msbCleared = counts & ~S{S::MostSignificantBit};
+        return S{static_cast<T>(msbCleared.value() << 1)};
+    };
+
+    exponent = S{static_cast<T>(exponent.value() << (NB - ActualBits))};
+    return associativeOperatorIterated_regressive(
+        x, S{1}, exponent, S{S::MostSignificantBit}, operation,
```

Comment:
Your neutral is, lane wise, { 0, ..., 1 }, so, the upper lanes are initialized to 0
You could `S{meta::BitmaskMaker<BaseType, 1, BitsPerLane>::value()}`

Reply:
ooh thanks for that, that makes total sense!

Reply:
Possibly, but using `broadcast` is a complication because it relies on multiplication.
I have been slowly internalizing what needs to be done to support SIMD: no assumptions about the availability of whole-base-type multiplication.  In ARM NEON, x86-64 SSE of 128 bits, there is no such thing, but at most lane-wise 64 bit multiplication, same with AVX 256 bits.

BitmaskMaker can be upgraded to support longer types.
The problem is that SIMD negate `constexpr`, it's going to give us pain to make the non-constexpr flavor of the library

Reply:
That is, indeed, subtle.

Perhaps not so subtle in the SIMD `constexpr` negation...